cypress/integration/patch/task-table.ts

```diff
@@ -0,0 +1,205 @@
+        .should("have.attr", "href")
+        .and("include", "/task");
+    });
+  });
+
+  it("Task count displays total tasks", () => {
+    cy.visit(pathTasks);
+    cy.waitForGQL("PatchTasks");
+    cy.get("[data-cy=total-task-count]").contains("47");
+  });
+
+  it("Sort buttons are disabled when fetching data", () => {
+    cy.visit(pathTasks);
+    cy.contains(TABLE_SORT_SELECTOR, "Name").click();
+    cy.once("fail", (err) => {
+      expect(err.message).to.include(
+        "'pointer-events: none' prevents user mouse interaction."
+      );
+    });
+  });
+
+  // flakey test - must fix
+  ["NAME", "STATUS", "BASE_STATUS", "VARIANT"].forEach((sortBy) => {
+    xit(`Fetches tasks sorted by ${sortBy} when ${sortBy} header is clicked`, () => {
+      clickSorterAndAssertTasksAreFetched(sortBy);
+    });
+  });
+
+  describe("Changing page number", () => {
+    beforeEach(() => {
+      cy.preserveCookies();
+    });
+
+    it("Displays the next page of results and updates URL when right arrow is clicked and next page exists", () => {
+      cy.visit(pathTasks);
+      clickOnPageBtnAndAssertURLandTableResults(
+        dataCyNextPage,
+        secondPageDisplayNames,
+        1,
+        dataCyTableRows
+      );
+    });
+
+    it("Displays the previous page of results and updates URL when the left arrow is clicked and previous page exists", () => {
+      clickOnPageBtnAndAssertURLandTableResults(
+        dataCyPrevPage,
+        firstPageDisplayNames,
+        0,
+        dataCyTableRows
+      );
+    });
+
+    it("Does not update results or URL when left arrow is clicked and previous page does not exist", () => {
+      clickOnPageBtnAndAssertURLandTableResults(
+        dataCyPrevPage,
+        firstPageDisplayNames,
+        0,
+        dataCyTableRows
+      );
+    });
+
+    it("Does not update results or URL when right arrow is clicked and next page does not exist", () => {
+      cy.visit(`${pathTasks}?page=4`);
+      clickOnPageBtnAndAssertURLandTableResults(
+        dataCyNextPage,
+        fourthPageDisplayNames,
+        4,
+        dataCyTableRows
+      );
+    });
+  });
+
+  describe("Changing page size updates URL and renders less than or equal to that many rows ", () => {
+    [20, 10, 50, 100].forEach((pageSize) => {
+      it(`Updates URL and displays up to ${pageSize} results at once when the page size is changed to ${pageSize}`, () => {
+        clickOnPageSizeBtnAndAssertURLandTableSize(
+          pageSize,
+          "[data-test-id=tasks-table-page-size-selector]",
+          `[data-test-id=tasks-table-page-size-selector-${pageSize}]`,
+          dataCyTableRows
+        );
+      });
+    });
+  });
+});
+
+const dataCyTableRows = "[data-test-id=tasks-table] tr td:first-child";
+const firstPageDisplayNames = [
+  "lint-service",
+  "test-cloud",
+  "test-service",
+  "test-thirdparty",
+  "test-thirdparty",
+  "test-model",
+  "generate-lint",
+  "js-test",
+  "test-agent",
+  "test-auth",
+];
```

Comment:
It does seem brittle but i think as long as we don't make updates the data this relies on it should be fine.  Since we are using this to assert the data we expect is being returned. This is the best way to determine the system actually operates correctly. If we had this project  in the same directory as the data we could have probably fetched it directly from the data we use to seed the db.

Reply:
I think it might make sense to move this test data outside of the test tho. Like how we do for the gql tests in evergreen. Let test data live in its own directory/file just to clean up these test files a little bit

Reply:
in this case i think it would be smart to have separate data sets for developing locally and for e2e tests. that way we can edit the local data freely without worrying about destroying e2e test results. created a ticket for this: https://jira.mongodb.org/browse/EVG-8009

Reply:
the data we use locally should closely resemble the data we use on the e2e tests because in the end the tests we write and run locally are ran on the e2e server it could get confusing if we use two seperate datasets locally and on the e2e server. Because one set of data might break and the other one wont.